### PR TITLE
feat: add support for OAuth credentials

### PR DESCRIPTION
**Pull request by pnadolny13**
Closes https://github.com/MeltanoLabs/tap-hubspot/issues/46

### CODE REVIEW

File: tap_hubspot/client.py [hunk 1]
```diff
@@ -47,12 +45,13 @@ def authenticator(self) -> _Auth:
...
-            self,
-            token=access_token,
-        )
+        if "refresh_token" in self.config:
+            return HubSpotOAuthAuthenticator.create_for_stream(stream=self)
+        else:
+            return BearerTokenAuthenticator.create_for_stream(
+                self,
+                token=self.config.get("access_token"),
+            )
```
Comment (edgarrmondragon) [inline tap_hubspot/client.py:54]:
```suggestion
            return BearerTokenAuthenticator(
                token=self.config.get("access_token"),
            )
```

File: tap_hubspot/auth.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+            "client_secret": self.config["client_secret"],
+            "refresh_token": self.config["refresh_token"],
+        }
+
+    @classmethod
+    def create_for_stream(cls, stream):
+        return cls(
+            stream=stream,
+            auth_endpoint="https://api.hubapi.com/oauth/v1/token",
+        )
```
Comment (edgarrmondragon) [inline tap_hubspot/auth.py:23]:
```suggestion
```

File: tap_hubspot/client.py [hunk 3]
```diff
@@ -47,12 +45,13 @@ def authenticator(self) -> _Auth:
...
         """
 
-        access_token = self.config.get("access_token")
-
-        return BearerTokenAuthenticator.create_for_stream(
-            self,
-            token=access_token,
-        )
+        if "refresh_token" in self.config:
+            return HubSpotOAuthAuthenticator.create_for_stream(stream=self)
```
Comment (edgarrmondragon) [inline tap_hubspot/client.py:49]:
```suggestion
            return HubSpotOAuthAuthenticator(
            
    auth_endpoint="https://api.hubapi.com/oauth/v1/token",
            )
```

**Review by edgarrmondragon — APPROVED** [review]

File: tap_hubspot/client.py (hunk 1 above)
Comment (pnadolny13) [reply tap_hubspot/client.py]: @edgarrmondragon with this change I get `TypeError: __init__() missing 1 required positional argument: 'stream'`. I think I still need to pass in the `self` as the first arg.

**Pull request merged**